### PR TITLE
The querry is now send to NLPAPI to filter it if intents matches the …

### PR DESCRIPTION
**Pull request by Atlantis251**
…on in appsettings.json

### CODE REVIEW

File: WhisperAPI/WhisperAPI/Services/SuggestionsService.cs [hunk 1]
```diff
@@ -40,6 +60,22 @@ public IEnumerable<SuggestedDocument> GetSuggestions(string querry)
             return documents;
         }
 
+        private bool IsIntentRelevant(NlpAnalysis nlpAnalysis)
+        {
+            foreach (var intent in nlpAnalysis.Intents)
```
Comment (Xahor) [inline WhisperAPI/WhisperAPI/Services/SuggestionsService.cs]: - Can't this be replaced by something like : `!nlpAnalysis.Intents.contains(this._intents)` ?
- Why do we check each (var intent in nlpAnalysis.Intents) ? Will we discard a querry if there is "need help" : 98% and "greeting" : 2% ?

File: WhisperAPI/WhisperAPI/Services/SuggestionsService.cs [hunk 2]
```diff
@@ -1,18 +1,38 @@
...
 namespace WhisperAPI.Services
 {
     public class SuggestionsService : ISuggestionsService
     {
         private readonly IIndexSearch _indexSearch;
 
-        public SuggestionsService(IIndexSearch indexSearch)
+        private readonly INlpCall _nlpCall;
+
+        private readonly List<string> _intents;
```
Comment (Xahor) [inline WhisperAPI/WhisperAPI/Services/SuggestionsService.cs]: - Not clear enough what ` List<string> _intents` is used for (are they relevant intent or intent that we want to discard?)

File: WhisperAPI/WhisperAPI/Settings/ApplicationSettings.cs [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
+
+namespace WhisperAPI.Settings
 {
     public class ApplicationSettings
     {
         public string ApiKey { get; set; }
+
+        public string NlpApiBaseAddress { get; set; }
+
+        public List<string> Intents { get; set; }
```
Comment (Xahor) [inline WhisperAPI/WhisperAPI/Settings/ApplicationSettings.cs]: Same as in "SuggestionsService" : Not clear enough what List<string> Intents is used for (are they relevant intent or intent that we want to discard?)

File: WhisperAPI/WhisperAPI.Tests/Unit/SuggestionsServiceTest.cs [hunk 4]
```diff
@@ -3,27 +3,117 @@
...
+            this._suggestionsService.GetSuggestions(querry).Should().BeEquivalentTo(new List<SuggestedDocument>());
+        }
+
+        [Test]
+        [TestCase("test")]
+        public void When_receive_intent_irrelevant_then_returns_empty_list_of_suggestedDocuments(string querry)
+        {
+            var intents = new List<Intent>
+            {
+                new IntentBuilder().WithName("Need help").Build(),
```
[the remark this answers is not in the document]
Comment (Atlantis251) [reply WhisperAPI/WhisperAPI.Tests/Unit/SuggestionsServiceTest.cs]: C'est voulu, c'est pour mettre un intent qui est bon dans la liste pour voir si ça va marcher pareil

File: WhisperAPI/WhisperAPI.Tests/Data/Builders/EntityBuilder.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        private string _name = "EntityName";
+
+        public EntityBuilder WithName(string name)
+        {
+            this._name = name;
+            return this;
+        }
+
+        public Entity Build()
```
[the remark this answers is not in the document]
Comment (Atlantis251) [reply WhisperAPI/WhisperAPI.Tests/Data/Builders/EntityBuilder.cs]: Done

File: WhisperAPI/WhisperAPI.Tests/Unit/SuggestionsServiceTest.cs [hunk 6]
```diff
@@ -3,27 +3,117 @@
...
+            this._nlpCallMock
+                .Setup(x => x.GetNlpAnalyses(It.IsAny<string>()))
+                .Returns(nlpAnalysis);
+
+            this._suggestionsService.GetSuggestions(querry).Should().BeEquivalentTo(new List<SuggestedDocument>());
+        }
+
+        [Test]
+        [TestCase("test")]
+        public void When_receive_intent_irrelevant_then_returns_empty_list_of_suggestedDocuments(string querry)
```
[the remark this answers is not in the document]
Comment (Atlantis251) [reply WhisperAPI/WhisperAPI.Tests/Unit/SuggestionsServiceTest.cs]: Done

File: WhisperAPI/WhisperAPI.Tests/Unit/SuggestionsServiceTest.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Atlantis251) [reply WhisperAPI/WhisperAPI.Tests/Unit/SuggestionsServiceTest.cs]: Done

File: WhisperAPI/WhisperAPI.Tests/Unit/SuggestionsServiceTest.cs [hunk 7]
```diff
@@ -94,46 +187,14 @@ public void SetUp()
...
-
-            this._suggestionsServiceValid = new SuggestionsService(this._indexSearchValid.Object);
-            this._suggestionsServiceEmpty = new SuggestionsService(this._indexSearchEmpty.Object);
-            this._suggestionsServiceNull = new SuggestionsService(this._indexSearchNull.Object);
         }
 
-        [Test]
-        [TestCase("test")]
-        public void When_receive_valid_searchresult_from_search_then_return_list_of_suggestedDocuments(string suggestion)
+        public List<string> GetIntentsNotRelevents()
```
[the remark this answers is not in the document]
Comment (Atlantis251) [reply WhisperAPI/WhisperAPI.Tests/Unit/SuggestionsServiceTest.cs]: Done

File: WhisperAPI/WhisperAPI/Services/INlpCall.cs [hunk 8]
```diff
@@ -0,0 +1,9 @@
+﻿using WhisperAPI.Models.NLPAPI;
+
+namespace WhisperAPI.Services
+{
+    public interface INlpCall
+    {
+        NlpAnalysis GetNlpAnalyses(string sentence);
```
[the remark this answers is not in the document]
Comment (Atlantis251) [reply WhisperAPI/WhisperAPI/Services/INlpCall.cs]: Done

File: WhisperAPI/WhisperAPI/Services/NlpCall.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+        public NlpCall(HttpClient httpClient, string baseAdress)
+        {
+            this._httpClient = httpClient;
+            this._baseAdress = baseAdress;
+        }
+
+        public NlpAnalysis GetNlpAnalyses(string sentence)
+        {
+            this._httpClient.BaseAddress = new Uri(this._baseAdress);
```
[the remark this answers is not in the document]
Comment (Atlantis251) [reply WhisperAPI/WhisperAPI/Services/NlpCall.cs]: Done

File: WhisperAPI/WhisperAPI/Services/SuggestionsService.cs (hunk 2 above)
Comment (Atlantis251) [reply WhisperAPI/WhisperAPI/Services/SuggestionsService.cs]: Done

File: WhisperAPI/WhisperAPI/Settings/ApplicationSettings.cs (hunk 3 above)
Comment (Atlantis251) [reply WhisperAPI/WhisperAPI/Settings/ApplicationSettings.cs]: Done

File: WhisperAPI/WhisperAPI/Services/SuggestionsService.cs (hunk 1 above)
Comment (Atlantis251) [reply WhisperAPI/WhisperAPI/Services/SuggestionsService.cs]: I agree with you about the discard. However, i think it's gonna be in another release or we can decide now what we want there

File: WhisperAPI/WhisperAPI/Services/NlpCall.cs [hunk 10]
```diff
@@ -35,5 +32,12 @@ private static StringContent CreateStringContent(string sentence)
         {
             return new StringContent($"{{\"sentence\": \"{sentence}\"}}", Encoding.UTF8, "application/json");
         }
+
+        private void InitHttpClient()
```
Comment (Xahor) [inline WhisperAPI/WhisperAPI/Services/NlpCall.cs]: As discusted, create a get/set and initialise in the get if not initialized

**Pull request merged**